packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -513,6 +513,9 @@ class ScrollableState<T extends Scrollable> extends State<T> with SingleTickerPr
   /// If there are no in-progress scrolling physics, this function scrolls to
   /// the given offset instead.
   void didUpdateScrollBehavior(double newScrollOffset) {
+    _contentExtent = scrollBehavior.contentExtent;
+    _containerExtent = scrollBehavior.containerExtent;
```

Comment:
This is tricky. The variables you're updating were added in https://github.com/flutter/flutter/pull/5862

If you actually change the _contentExtent here then you'd need to setState(). But I believe that if you wrap the changes to _contentExtent and _containerExtent in setState() you'll trip over an assertion.